notebooks/C_CNN_Training_and_Validation.ipynb

```diff
@@ -1,602 +1,602 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "6Q72wbtUm1qD"
+   },
+   "source": [
+    "### Install the CellX library & create subdirectories in the virtual machine:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "hOtdYdHfm1qD"
+   },
+   "outputs": [],
+   "source": [
+    "# if using colab, install cellx library and make log and data folders\n",
+    "\n",
+    "if 'google.colab' in str(get_ipython()):\n",
+    "    try: \n",
+    "        import cellx\n",
+    "    except:\n",
+    "        !pip install -q git+git://github.com/quantumjot/cellx.git\n",
+    "        !mkdir logs\n",
+    "        !mkdir train\n",
+    "        !mkdir test"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "JxGN9I8Tm1qE"
+   },
+   "source": [
+    "### Import libraries and CellX toolkit:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "jI0vgg9jm1qE"
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import zipfile\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "from datetime import datetime\n",
+    "from skimage.transform import resize"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "jSQ2HRVJm1qE"
+   },
+   "outputs": [],
+   "source": [
+    "import tensorflow.keras as K\n",
+    "import tensorflow as tf"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "X4FcQHy6m1qE"
+   },
+   "outputs": [],
+   "source": [
+    "from cellx.layers import Encoder2D\n",
+    "from cellx.tools.dataset import build_dataset\n",
+    "from cellx.tools.dataset import write_dataset\n",
+    "from cellx.augmentation.utils import append_conditional_augmentation, augmentation_label_handler\n",
+    "from cellx.callbacks import tensorboard_confusion_matrix_callback\n",
+    "from cellx.tools.io import read_annotations"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Xc6DAoZ3m1qF"
+   },
+   "source": [
+    "### Define paths & class labels:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "Rw1ipBL0m1qF"
+   },
+   "outputs": [],
+   "source": [
+    "TRAIN_PATH = \"./train\"\n",
+    "TEST_PATH = \"./test\"\n",
+    "TRAIN_FILE = os.path.join(TRAIN_PATH, 'CNN_train.tfrecord')\n",
+    "TEST_FILE = os.path.join(TEST_PATH, 'CNN_test.tfrecord')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Hv5QAp3um1qF"
+   },
+   "source": [
+    "### Set-up CNN training hyperparameters: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "u-QTgXl7m1qF"
+   },
+   "outputs": [],
+   "source": [
+    "BATCH_SIZE = 64\n",
+    "BUFFER_SIZE = 20_000\n",
+    "TRAINING_EPOCHS = 100\n",
+    "BOUNDARY_AUGMENTATION = True"
```

Comment:
Probably worth adding:
`INPUT_SHAPE = (64, 64, 1)`

Reply:
ah ok, I thought about it but wasn't sure that's what you had in mind! will implement it now